cmd/supervise/main.go

```diff
@@ -1,13 +1,21 @@
 package main
 
 import (
+	"encoding/json"
 	"os"
 
 	"github.com/deref/exo/internal/supervise"
+	"github.com/deref/exo/internal/util/cmdutil"
 )
 
 func main() {
-	command := os.Args[0]
-	args := os.Args[1:]
-	supervise.Main(command, args)
+	cfg := &supervise.Config{}
+	if err := json.NewDecoder(os.Stdin).Decode(cfg); err != nil {
```

Comment:
why duplicate this code rather than stick it inside `supervise.Main`?

Reply:
I don't want supervise.Main to depend on reading from `os.Stdin`, and it is called from exop and exod, which do not expect input on stdin.

Reply:
I'm confused about this. Seems like there is some leftover dev stuff that is marked "XXX" and "This is broken". Should we remove/cleanup that code?

Regarding `exop` - I'm not sure why that code would need a `supervise` codepath at all. I don't remember adding that. There is already the `supervise/main.go` codepath for testing the supervise script directly without the exo binary. Integrating it in to exop seems redundant.

As for the exod codepath, I think that was a quick and dirty hack to deal with the fact that exod wants to launch supervise by looking for `argv[0]` to spawn a new process. Now that we have a proper config mechanism, maybe we should use that instead to supply a supervise path?